### PR TITLE
Fix #214

### LINKED ISSUE #214: Auto rename fastq files for cellranger input
### Description of feature

Cellranger requires the fastq files to follow a specific pattern. It would be nice to have the pipeline taking care of renaming the files.
One possibility would be to create a symlink for each file with the required pattern using regex.

### PR DESCRIPTION
**Pull request by sandrocarollo**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/scrnaseq/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/scrnaseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: modules/local/samplesheet_check.nf [hunk 1]
```diff
@@ -22,7 +22,11 @@ process SAMPLESHEET_CHECK {
     check_samplesheet.py \\
         $samplesheet \\
         samplesheet.valid.csv
-
+    if("${params.aligner}"=="cellranger") {
+        samplesheet_cellranger_conform.py \\
+        samplesheet.valid.csv \\
+        samplesheet_cellranger.csv
+    }
```
Comment (grst) [inline modules/local/samplesheet_check.nf:29]: The `if` here would need to be written in `bash`. 

Alternatively, you put the `if` outside the `"""` block and assign the result in a variable that can be included in the scripts block.

**Comment by grst** [comment]
@vjmarteau  @sandrocarollo, this probably needs to become a separate process. 

Currently, the symbolic links would be created in the work directory of the process that writes the samplesheet, but the downstream process would not have access to them. The created symlinks need to be part of some `output` channel in nextflow, otherwise they are not accessible from the next processes.

**Comment by grst** [comment]
That process could consume the `ch_fastq` here
https://github.com/nf-core/scrnaseq/blob/c6ec2175a8f5d593c00136835c45c6291821fe11/subworkflows/local/align_cellranger.nf#L37

and emit a new `ch_fastq` with the fixed filenames.

**Comment by vjmarteau** [comment]
> @vjmarteau @sandrocarollo, this probably needs to become a separate process.
> 
> Currently, the symbolic links would be created in the work directory of the process that writes the samplesheet, but the downstream process would not have access to them. The created symlinks need to be part of some `output` channel in nextflow, otherwise they are not accessible from the next processes.

I think that would also solve the issue when the fastq files already have the right pattern. Right now it will fail because a file of this name already exists.

**Comment by grst** [comment]
Closing in favor of #246.

**Pull request closed without merging**